src/localconf.js

```diff
@@ -1,155 +1,206 @@
-  submissionapi_oauth_path = hostname + 'api/v0/oauth2/';
-  credential_path = hostname + 'middleware/aws/v0/access_key/';
-  credential_oauth_path = hostname + 'middleware/oauth2/v0/';
-  credential_cdis_path = userapi_path + 'credentials/cdis/';
-  graphql_path = hostname + 'api/v0/submission/graphql/';
-  graphql_schema_url = hostname + '/data/schema.json';
-  appname = 'The Brain Commons Portal';
-  nav_items = [
-    {'icon': 'home', 'link': '/', 'color': '#A51C30', 'name': 'home'},
-    {'icon': 'search', 'link': '/query', 'color': '#2D728F', 'name': 'query'},
-    {'icon': 'class', 'link': '/DD', 'color': '#A51C30', 'name': 'dictionary'},
-    {'icon': 'face', 'link': '/identity', 'color': '#2D728F', 'name': 'profile'},
-    {'icon': 'content_copy', 'link': '/files', 'color': '#a2a2a2', 'name': 'data'}
-  ];
-  login = {
-    url: userapi_path + 'login/google' + '?redirect=',
-    title: 'Login from Google'
-  };
-} else if (app === 'acct'){
-  required_certs = [];
-  userapi_path = hostname + 'user/';
-  submissionapi_path = hostname + 'api/v0/submission/';
-  submissionapi_oauth_path = hostname + 'api/v0/oauth2/';
-  credential_path = hostname + 'middleware/aws/v0/';
-  credential_oauth_path = hostname + 'middleware/oauth2/v0/';
-  credential_cdis_path = userapi_path + 'credentials/cdis/';
-  graphql_path = hostname + 'api/v0/submission/graphql/';
-  graphql_schema_url = hostname + '/data/schema.json';
-  appname = 'ACCOuNT Data Commons Portal';
-  nav_items = [
-    {'icon': 'home', 'link': '/', 'color': '#a2a2a2', 'name': 'home'},
-    {'icon': 'search', 'link': '/query', 'color': '#daa520', 'name': 'query'},
-    {'icon': 'class', 'link': '/DD', 'color': '#a2a2a2', 'name': 'dictionary'},
-    {'icon': 'face', 'link': '/identity', 'color': '#daa520', 'name': 'profile'},
-    {'icon': 'content_copy', 'link': '/files', 'color': '#a2a2a2', 'name': 'data'}
-  ];
-  login = {
-    url: userapi_path + 'login/google' + '?redirect=',
-    title: 'Login from Google'
-  };
-} else if (app ==='gdc') {
-  userapi_path = dev === true ? hostname + 'user/' : hostname + 'api/';
-  credential_path = userapi_path + 'credentials/cleversafe/';
-  credential_oauth_path = userapi_path + 'oauth2/';
-  credential_cdis_path = userapi_path + 'credentials/cdis/';
-  graphql_schema_url = hostname + 'data/schema.json';
-  appname = 'GDC Jamboree Portal';
-  nav_items = [
-    {'icon': 'home', 'link': '/', 'color': '#a2a2a2', 'name': 'home'}
-  ];
-  login = {
-    url: 'https://itrusteauth.nih.gov/affwebservices/public/saml2sso?SPID=https://bionimbus-pdc.opensciencedatacloud.org/shibboleth&RelayState=',
-    title: 'Login from NIH'
+  let userapiPath, submissionApiPath, submissionApiOauthPath, credentialPath, credentialOauthPath,
+    credentialCdisPath, graphqlPath, appname, navItems, login, graphqlSchemaUrl;
+  let requiredCerts = [];
+
+  // Support data/gqlSetup.js auto-generation of data-portal graphQL
+  let gqlSetup = {
+    fileTypeList: [
```

Comment:
can this be parsed from dictionary?
our goal is we don't need to update anything when we setup new commons

Reply:
Yeah - we can 'curl' the dictionary from http://localhost/api/v0/submission/_dictionary/_all , but then we have a dependency on an external service.

As a practical matter we already have to edit localconf.js to deploy a new commons.

Either way is messy.

Anyway - this is a first step ...

Reply:
current editting in localconf for new commons is mainly about portal related configurations that's not described in dictionary. 
The data related configuration though is already represented in dictionary, duplicating the same info here would be error prone and also means updating dictionary for a commons every time need to update portal code.
We already have this external service dependency to get schema in data folder,  I think it's an 
 innevitable dependency? either automatically get the info from the external service or manually map the info into portal code and make sure it's in sync?